discovery/github-iterator.go

```diff
@@ -92,16 +99,20 @@ func newGithubClient(token string, timeout time.Duration) *github.Client {
 func (p *GHOrgReposIter) Next(
 	ctx context.Context,
 ) (*github.Repository, time.Duration, error) {
-	if len(p.repos) == 0 {
-		retry, err := p.requestRepos(ctx)
-		if err != nil && len(p.repos) == 0 {
-			return nil, retry, err
+	for {
+		if len(p.repos) == 0 {
+			retry, err := p.requestRepos(ctx)
+			if err != nil && len(p.repos) == 0 {
+				return nil, retry, err
+			}
 		}
-	}
 
-	var next *github.Repository
-	next, p.repos = p.repos[0], p.repos[1:]
-	return next, 0, nil
+		var next *github.Repository
+		next, p.repos = p.repos[0], p.repos[1:]
+		if !p.excludedRepos[next.GetName()] {
```

Comment:
maps in Go returns two values, one with the value for the given key and a boolean that tells you if that key is present in the map
```go
m := map[string]int{
        "foo": 1,
}

v, ok := m["foo"] // v = 1, ok = true
v, ok := m["bar"] // v = default zero value for the type, ok = false
```
It would be more idiomatic check the boolean returned to check if the key is in the map.

Also, when you are using a map as a set sometimes the definition `map[string]struct{}` for the map is used so you can see at a first sight that is intended to be a set.
```go
var m map[string]struct{}
m := map[string]struct{}{}
m["foo"] = struct{}{}
_, ok := m["foo"] // we don't care about the value, ok = true
_, ok := m["bar"] // we don't care about the value, ok = false

```

Reply:
OK, made the change, I'm not sure if it will be 100% idiomatic yet, though